lib/src/iterable_extensions.dart

```diff
@@ -914,6 +914,18 @@ extension IterableIterableExtension<T> on Iterable<Iterable<T>> {
       };
 }
 
+/// Extensions that apply to iterables of tuples/pairs (records with two unnamed elements).
+extension IterableTupleExtension<T1, T2> on Iterable<(T1, T2)> {
```

Comment:
I'm not keen on just using a positional pair as the representation of a key/value pair.
I'd use a type like `({T1 key, T2 value})` instead.
It does get a little more verbose if you don't want to use `key` and `value` as the names for the key and value.

Maps are not sets of pairs. (Now that we have records, `Set<(...,...)>` is a sets of pairs!)